libcxx/utils/ci/buildkite-pipeline.yml

```diff
@@ -28,916 +28,509 @@ env:
-        ENABLE_CLANG_TIDY: "On"
-    agents:
-      queue: "libcxx-builders"
-      os: "linux"
-    retry:
-      automatic:
-        - exit_status: -1  # Agent was lost
-          limit: 2
-    timeout_in_minutes: 120
+# Define common YAML anchors to reduce duplication
+definitions:
+  _linux_agent: &linux_agent
+      agents:
+        queue: "libcxx-builders"
+        os: "linux"
 
-  - label: "Modular build"
-    command: "libcxx/utils/ci/run-buildbot generic-modules"
-    artifact_paths:
-      - "**/test-results.xml"
-      - "**/*.abilist"
-    env:
-        CC: "clang-${LLVM_HEAD_VERSION}"
-        CXX: "clang++-${LLVM_HEAD_VERSION}"
-        ENABLE_CLANG_TIDY: "On"
-    agents:
-      queue: "libcxx-builders"
-      os: "linux"
-    retry:
-      automatic:
-        - exit_status: -1  # Agent was lost
-          limit: 2
-    timeout_in_minutes: 120
+  _windows_agent: &windows_agent
+      agents:
+        queue: "windows"
 
-  - label: "C++11"
-    command: "libcxx/utils/ci/run-buildbot generic-cxx11"
-    artifact_paths:
-      - "**/test-results.xml"
-      - "**/*.abilist"
-    env:
-        CC: "clang-${LLVM_HEAD_VERSION}"
-        CXX: "clang++-${LLVM_HEAD_VERSION}"
-        ENABLE_CLANG_TIDY: "On"
-    agents:
-      queue: "libcxx-builders"
-      os: "linux"
+  _mac_agent: &mac_agent
+      agents:
+        queue: "libcxx-builders"
+        os: "macos"
+        arch: "x86_64" # TODO: Remove this once we are able to run back-deployment on arm64 again, since this isn't x86_64 specific
+
+  _common: &common
+    timeout_in_minutes: 120
     retry:
       automatic:
         - exit_status: -1  # Agent was lost
           limit: 2
-    timeout_in_minutes: 120
-
-  - label: "C++03"
-    command: "libcxx/utils/ci/run-buildbot generic-cxx03"
     artifact_paths:
       - "**/test-results.xml"
       - "**/*.abilist"
-    env:
-        CC: "clang-${LLVM_HEAD_VERSION}"
-        CXX: "clang++-${LLVM_HEAD_VERSION}"
-        ENABLE_CLANG_TIDY: "On"
-    agents:
-      queue: "libcxx-builders"
-      os: "linux"
-    retry:
-      automatic:
-        - exit_status: -1  # Agent was lost
-          limit: 2
-    timeout_in_minutes: 120
+      - "**/crash_diagnostics/*"
+
+  _common_env: &common_env
+      ENABLE_CLANG_TIDY: "On"
+      LLVM_SYMBOLIZER_PATH: "/usr/bin/llvm-symbolizer-${LLVM_HEAD_VERSION}"
+      CLANG_CRASH_DIAGNOSTICS_DIR: "crash_diagnostics"
+      CC: clang-${LLVM_HEAD_VERSION}
+      CXX: clang++-${LLVM_HEAD_VERSION}
+      CMAKE: /opt/bin/cmake
+
+  _versioned_clang: &versioned_clang
+    # Note modules require and absolute path for clang-scan-deps
+    # https://github.com/llvm/llvm-project/issues/61006
+    CC: /usr/lib/llvm-${LLVM_HEAD_VERSION}/bin/clang
+    CXX: /usr/lib/llvm-${LLVM_HEAD_VERSION}/bin/clang++
+
+
+steps:
+- label: GCC ${GCC_STABLE_VERSION} / C++latest
+  command: libcxx/utils/ci/run-buildbot generic-gcc
+  <<: *common
+  <<: *linux_agent
+  env:
+    <<: *common_env
+    CC: gcc-${GCC_STABLE_VERSION}
```

Comment:
Just to validate my understanding -- this `CC: ...` overrides the one present in `common_env? Should we instead have anchors for specific compiler versions and avoid including them into `common_env`?

Reply:
Yep, that's how it works.

I think both approaches have benefits. 
Specifying the compiler separately is more explicit, which is nice.
But defaulting to Clang++ seems better than c++ if the job omits the compiler variables.

I'll go with your suggestion.

Reply:
Reopening because I said one thing and did another. FYI.